hicommonwealth/commonwealth
# React Quill Markdown

Adds markdown support to the `ReactQuillEditor` component and updates parent components to support the new functionality via serialization/deserialization of text.

## Link to Issue
Closes: #2933

## Description of Changes
- Adds M/R (Markdown/Richtext) button to editor to switch between modes
- Adds preview button which opens preview modal on click
- Updates parent components to use serialize/deserialize logic which helps with parsing markdown vs richtext

## Test Plan
- Create a new comment (richtext by default)– confirm that it formats and submits correctly
- Create a new comment and switch to markdown mode– confirm that it formats and submits correctly
- Create a new comment, switch between the modes, and click the preview button– confirm the preview reflects the current mode
- Click Edit on a richtext comment– confirm that the editor opens in richtext mode
- Click Edit on a markdown comment– confirm that the editor opens in markdown mode
- Other affected areas include: Create Topic, Edit Topic, Create Thread, Edit Thread

## Deployment Plan
N/A

## Other Considerations
- Currently, the editor toolbar buttons don't work correctly in markdown mode, but markdown support otherwise works correctly
